pkg/operator/v1helpers/helpers.go

```diff
@@ -58,7 +58,7 @@ func SetOperatorCondition(conditions *[]operatorv1.OperatorCondition, newConditi
 
 	if existingCondition.Status != newCondition.Status {
 		existingCondition.Status = newCondition.Status
-		existingCondition.LastTransitionTime = newCondition.LastTransitionTime
+		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
```

Comment:
@deads2k this was resulting in a nil transition time when updating an existing condition.